src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -19,62 +20,46 @@
 
-    public static PineconeConnection createIndexIfNotExistsDataPlane(int dimension, String indexType) throws IOException, InterruptedException {
+    public static AbstractMap.SimpleEntry<String, Pinecone> createIndexIfNotExistsDataPlane(int dimension, String indexType) throws IOException, InterruptedException {
         String apiKey = System.getenv("PINECONE_API_KEY");
         Pinecone pinecone = new Pinecone(apiKey);
 
         String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
-        if (indexName.isEmpty()) indexName = createNewIndex(pinecone, dimension, indexType);
+        if (indexName.isEmpty()) indexName = createNewIndex(pinecone, dimension, indexType, true);
 
-        // Do not proceed until the newly created index is ready
-        isIndexReady(indexName, pinecone);
-
-        // Adding to test PineconeConnection(pineconeConfig, host) constructor
-        String host = pinecone.describeIndex(indexName).getHost();
-        PineconeConfig config = new PineconeConfig(apiKey);
-        config.setHost(host);
-        return new PineconeConnection(config);
-    }
-
-    public static String createIndexIfNotExistsControlPlane(Pinecone pinecone, int dimension, String indexType) throws IOException, InterruptedException {
-        String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
-
-        return (indexName.isEmpty()) ? createNewIndex(pinecone, dimension, indexType) : indexName;
+        return new AbstractMap.SimpleEntry<>(indexName, pinecone);
     }
 
-    public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimension, String indexType)
-            throws InterruptedException {
+    public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimension, String indexType) {
         String indexName = "";
-        int i = 0;
         List<IndexModel> indexModels = pinecone.listIndexes().getIndexes();
         if(indexModels == null) {
             return indexName;
         }
-        while (i < indexModels.size()) {
-             IndexModel indexModel = isIndexReady(indexModels.get(i).getName(), pinecone);
+
+        for (IndexModel indexModel : indexModels) {
             if (indexModel.getDimension() == dimension
-                    && ((indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD)
-                        && indexModel.getSpec().getPod() != null
-                        && indexModel.getSpec().getPod().getReplicas() == 1
-                        && indexModel.getSpec().getPod().getPodType().equalsIgnoreCase("p1.x1"))
-                    || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS)))) {
+                && (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD) && indexModel.getSpec().getPod() != null)
+                || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS) && indexModel.getSpec().getServerless() != null)
+            ) {
```

Comment:
I was seeing issues related to this function finding indexes for the API key that were unrelated to the test that was running. I wanted to make this a bit easier to reason about, and I think we should eventually phase it out.